src/common/types/hash.cpp

```diff
@@ -84,7 +84,7 @@ hash_t HashBytes(const_data_ptr_t ptr, const idx_t len) noexcept {
 	// Hash/combine in blocks of 8 bytes
 	const auto remainder = len & 7U;
 	for (const auto end = ptr + len - remainder; ptr != end; ptr += 8U) {
-		h ^= Load<hash_t>(ptr);
+		h ^= BSwapIfBE(Load<hash_t>(ptr));
```

Comment:
Maybe we can have a dedicated method for this, e.g. `LoadLE<>` or `StoreLE<>` that includes a byte swap if on a BE system?